lib/workers/repository/config-migration/branch/commit-message.ts

```diff
@@ -4,43 +4,61 @@ import type { CommitMessage } from '../../model/commit-message';
 import { CommitMessageFactory } from '../../model/commit-message-factory';
 
 export class ConfigMigrationCommitMessageFactory {
-  private readonly config: RenovateConfig;
+  private commitMessage: string | null = null;
+  private prTitle: string | null = null;
 
-  private readonly configFile: string;
+  constructor(
+    private readonly config: RenovateConfig,
+    private readonly configFile: string
+  ) {}
 
-  constructor(config: RenovateConfig, configFile: string) {
-    this.config = config;
-    this.configFile = configFile;
-  }
-
-  create(): CommitMessage {
+  private create(isTitle = false): CommitMessage {
+    const messageTopic = isTitle
+      ? `Migrate renovate config`
+      : `Migrate config ${this.configFile}`;
     const { commitMessage } = this.config;
+    let { commitMessageAction, commitMessageTopic } = this.config;
 
-    this.config.commitMessageAction =
-      this.config.commitMessageAction === 'Update'
-        ? ''
-        : this.config.commitMessageAction;
+    commitMessageAction =
+      commitMessageAction === 'Update' ? '' : commitMessageAction;
 
-    this.config.commitMessageTopic =
-      this.config.commitMessageTopic === 'dependency {{depName}}'
-        ? `Migrate config ${this.configFile}`
-        : this.config.commitMessageTopic;
+    commitMessageTopic =
+      commitMessageTopic === 'dependency {{depName}}'
+        ? messageTopic
+        : commitMessageTopic;
 
-    this.config.commitMessageExtra = '';
-    this.config.semanticCommitScope = 'config';
+    const config = {
+      ...this.config,
+      semanticCommitScope: 'config',
+      commitMessageExtra: '',
+      commitMessageAction,
+      commitMessageTopic,
+    };
 
-    const commitMessageFactory = new CommitMessageFactory(this.config);
+    const commitMessageFactory = new CommitMessageFactory(config);
     const commit = commitMessageFactory.create();
 
     if (commitMessage) {
-      commit.subject = template.compile(commitMessage, {
-        ...this.config,
-        commitMessagePrefix: '',
-      });
+      config.commitMessagePrefix = '';
+      commit.subject = template.compile(commitMessage, config);
     } else {
-      commit.subject = `Migrate config ${this.configFile}`;
+      commit.subject = messageTopic;
     }
 
     return commit;
   }
+
+  getCommitMessage(): string {
```

Comment:
Note that these 2 funcitons getCommitMessage() and getPrTitle are exactly the same with different input, 
it could be 1 function with different input, could be a string or an enum or as i stated below could be the 
title string 
`Migrate renovate config` or `Migrate config ${this.configFile}`

Reply:
i dont want to expose any "logic" outside this class. 
this class either return a commit message or a pr title.
so the public methods are only wrappers that encapsulate the input for the private method `create` and are basically the api to this class.

Reply:
not a big deal then